keps/sig-node/4004-deprecate-kube-proxy-version/README.md

```diff
@@ -0,0 +1,265 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecate the `status.nodeInfo.kubeProxyVersion` field of v1.Node
+
+## Motivation
+
+This field is not accurate, the field is set by kubelet, which does not actually know the kube-proxy version,
+or even if kube-proxy is running.
+
+### Goals
+
+- Mark `status.nodeInfo.kubeProxyVersion` deprecated.
+- Find any places use it and get them to stop.
+- Make kubelet stop setting the field.
+
+### Non-Goals
+
+## Proposal
+
+- The proposal is to deprecate the kubeProxyVersion field on NodeStatus, which would affect GCP cloud providers (e.g. https://github.com/kubernetes/cloud-provider-gcp/pull/533, https://github.com/kubernetes/kubernetes/pull/117806) or other components, so we will use a feature gates to protect it until we fix the GCP cloud provider and any other components we find using the deprecated fields
+
+### Risks and Mitigations
+
+## Design Details
+
+### Versioned API Change: NodeStatus v1 core
+
+Mark the kubeProxyVersion field as Deprecated.
+
+```
+// Deprecated: KubeProxy Version reported by the node.
+KubeProxyVersion string `json:"kubeProxyVersion" protobuf:"bytes,8,opt,name=kubeProxyVersion"`
+```
+
+### NodeStatus Internal Representation
+
+Mark the kubeProxyVersion field as Deprecated.
+
+```
+// Deprecated: KubeProxy Version reported by the node.
+KubeProxyVersion string
+```
+
+### Test Plan
+
+[x ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+- Add a test to `TestVersionInfo` in `pkg/kubelet/nodestatus` to see if FeatureGate behaves as expected when it is turned on or off.
+- If DisableNodeKubeProxyVersion FeatureGate is enabled:
+  - `status.nodeInfo.kubeProxyVersion` will be empty.
+- Else:
+  - `status.nodeInfo.kubeProxyVersion` will be not empty .
+
+##### Integration tests
+
+
+- N/A
+
+##### e2e tests
+
+ - N/A
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Created the feature gate, disabled by default.
+- [ ] Started looking for components that might be using the deprecated field.
+
+#### Beta
+
+- [ ] Make sure that we fixed the GCP cloud provider and any other components we found that use the deprecated field.
```

Comment:
GCP cloud provider is already done at this point so that doesn't need to be mentioned here

Reply:
removed